### ISSUE TITLE
Use in project

### ISSUE THREAD

**Issue by CryptoDevving** [emilamaj/matchl1#1]
May I use this in my crypto currency project? What is the license rights?

**Comment by emilamaj** [comment]
I haven't set a license yet, so feel free to use any part of the code in the repo as you'd like.
I am showcasing signswap to find new freelance missions in crypto. Tell me if you need help on your crypto project

**Comment by CryptoDevving** [comment]
Thank you so much. I only have one question. How would I be able to add my own tokens to the swap? I am using this on a custom Polygon Edge deployment.

**Comment by emilamaj** [comment]
You mean to add your own tokens as a regular user or, as the deployer wanting to provide liquidity to match an incoming order ?

**Comment by CryptoDevving** [comment]
Both ways. As the deployer to add default listings primarily. So for example we would have Extella (gas coin) and then ERC20Token1 or ERC20Token2 and ERCToken3. But they are default listings on the app. They would be default on the drop down list.

**Comment by emilamaj** [comment]
As of now, the backend can automatically take a flashloan from Uniswap to match an incoming user order, and repay the loan with the wiggle room in the price set by the user (actually that part is 95% done)
Some changes needs to be made so that the backend would use it's own source of liquidity to fill in the user's order.

**Comment by CryptoDevving** [comment]
But I am not using uniswap, I am deploying this on my own blockchain.

**Comment by emilamaj** [comment]
Yes, I guess so then. You'll also need to create a different internal trade smart contract

**Comment by CryptoDevving** [comment]
I deleted my last comment, but thank you for all your help and understanding.
